base/src/test/java/org/mozilla/jss/tests/GenerateTestCert.java

```diff
@@ -303,8 +303,13 @@ private Certificate makeCert(String issuerName,
         int rand,
         SEQUENCE extensions) throws Exception {
 
-        AlgorithmIdentifier sigAlgID = new AlgorithmIdentifier(sigAlg.toOID());
-
+        AlgorithmIdentifier sigAlgID = null;
+        if(keyType.equals("RSA")) {
+            sigAlgID = new AlgorithmIdentifier(sigAlg.toOID(), null);
```

Comment:
Just a note that while this is true for rsaEnc OID'd certificates (needing the explicit `null`), the opposite holds for RSA-PSS-oid certificates if you end up wanting to use them in the test suite.

Reply:
I notices this comment after the merge, very sorry!

Actually, the class `Certificate` create the AlgorithmIdentifier with `null` if the algorithm is RSA. We have to make them consistent and I think we could remove the `null` in the `Certificate` and it should work. However, I prefered to modify only the test code . 

If we would add tests with RSA-PSS-oid  certificate the class should be modified because they should have parameters (I did a quick look to the spec) so some change is needed in any case, is this right?